concourse/scripts/create_concourse_cert.sh

```diff
@@ -19,15 +40,39 @@ if [ "${cert_status}" = "ISSUED" ]; then
   exit 0
 fi
 
-if [ "${created_cert}" = "false" ]; then
-  echo "Resending validation email for ${concourse_fqdn} certificate"
-  aws acm resend-validation-email --certificate-arn "${arn}" --domain "${concourse_fqdn}" --validation-domain "${SYSTEM_DNS_ZONE_NAME}"
+# The validation records are not returned for a couple of seconds from the API after creation
+echo "Getting DNS validation records"
+for _ in $(seq 20); do
+  sleep 3
+
+  cert_info=$(aws acm describe-certificate --certificate-arn "${arn}" --query 'Certificate')
+  dns_validation_record=$(echo "${cert_info}" | jq -r '.DomainValidationOptions[0].ResourceRecord.Name')
+  dns_validation_value=$(echo "${cert_info}" | jq -r '.DomainValidationOptions[0].ResourceRecord.Value')
+
+  if [ "null" != "${dns_validation_record}" ] && [ "null" != "${dns_validation_value}" ]; then
+    break
+  fi
+done
+
+if [ "null" = "${dns_validation_record}" ] || [ "null" = "${dns_validation_value}" ]; then
+  echo "DNS validation records are not found in the AWS API response: ${cert_info}"
+  echo
+  echo "Please run the script again"
+  exit 1
 fi
 
+zone_id=$(aws route53 list-hosted-zones-by-name --dns-name dev.cloudpipeline.digital --query 'HostedZones[0].Id' --output text | cut -d '/' -f 3)
```

Comment:
It doesn't look right to have `dev.cloudpipeline.digital` hardcoded here. How is this going to work in non-dev environments?

Reply:
You are right. It won't cause a problem in non-dev environments as we already have valid certificates there. I'll create a fix for it.